src/webgpu/shader/execution/expression/expression.ts

```diff
@@ -1059,3 +1059,79 @@ export function generateBinaryToU32Cases(
     return cases;
   }, new Array<Case>());
 }
+
+/**
+ * A function that performs a binary operation on x and y, and returns the expected
+ * result.
+ */
+export interface BinaryOp {
+  (x: number, y: number): number;
```

Comment:
For now, this only returns number. Will be changing this to return number|undefined when implementing division and modulo of scalar and vector.